libqtile/scripts/check.py

```diff
@@ -119,16 +120,22 @@ def check_config(args):
 
         # are the top level config variables the right type?
         module_name = path.splitext(path.basename(args.configfile))[0]
-        type_check_config_vars(tempdir, module_name)
+        ok &= type_check_config_vars(tempdir, module_name)
```

Comment:
Don't see that operator too often! Is simplicity better than readability?

Reply:
The thing here is that we want to all of these checks but store whether any have failed to print a message and return with a non-0 error code. Is there a more readable way to do this? Happy to change it.

Something like this?:
```diff
diff --git a/libqtile/scripts/check.py b/libqtile/scripts/check.py
index 2761cb61..defde918 100644
--- a/libqtile/scripts/check.py
+++ b/libqtile/scripts/check.py
@@ -109,7 +109,7 @@ def check_deps() -> None:
 def check_config(args) -> None:
     check_deps()
 
-    ok = True
+    failed = False
     print("Checking Qtile config at:", args.configfile)
 
     # need to do all the checking in a tempdir because we need to write stuff
@@ -120,10 +120,12 @@ def check_config(args) -> None:
 
         # are the top level config variables the right type?
         module_name = path.splitext(path.basename(args.configfile))[0]
-        ok &= type_check_config_vars(tempdir, module_name)
+        if not type_check_config_vars(tempdir, module_name):
+            failed = True
 
         # are arguments passed to qtile APIs correct?
-        ok &= type_check_config_args(tmp_path)
+        if not type_check_config_args(tmp_path):
+            failed = True
 
     # can we load the config?
     config = confreader.Config(args.configfile)
@@ -134,7 +136,7 @@ def check_config(args) -> None:
         print(e)
         print("\nYour config cannot be loaded by Qtile.")
 
-    if not ok:
+    if failed:
         sys.exit(1)
 
 
```

Reply:
That's certainly more readable but I'm happy to stick with your original.